src/code/speed_meter.c

```diff
@@ -1,25 +1,60 @@
 #include "global.h"
 #include "vt.h"
 
-volatile OSTime D_8016A520;
-volatile OSTime D_8016A528;
-volatile OSTime D_8016A530;
-volatile OSTime D_8016A538;
-volatile OSTime D_8016A540;
-volatile OSTime D_8016A548;
-volatile OSTime D_8016A550;
-volatile OSTime D_8016A558;
-volatile OSTime gRSPAudioTotalTime;
-volatile OSTime gRSPGFXTotalTime;
-volatile OSTime gRSPOtherTotalTime;
+/**
+ * How much time the audio update on the audio thread (`func_800E4FE0`) took in total, between scheduling the last two
+ * graphics tasks.
+ */
+volatile OSTime gAudioThreadUpdateTimeTotalPerGfxTask;
+/**
+ * How much time elapsed between scheduling the previous graphics task and the current one being ready (the previous
+ * task not necessarily being finished yet), without the amount of time spent on the audio update in the audio thread.
+ */
+volatile OSTime gGfxTaskSentToNextReadyMinusAudioThreadUpdateTime;
```

Comment:
Imo the "without the amount of time spent on audio" should either be left out of the name or reworded to something that's not a full description in the name itself (tho I don't have a proper suggestion for that atm)

Reply:
Yeah the name is horrible I agree, but it's very important though that the "time on audio" is not accounted in that timer as it's a big chunk,
![image](https://user-images.githubusercontent.com/5306619/193255613-822c865d-228d-4b19-98c0-ab81026b4c57.png)

Each chunk (1/4th) of the top blue-green-red-magenta bar represents the time for a VI retrace (see `SpeedMeter_DrawTimeEntries`) i.e. 1/50 seconds in PAL

first timer bar (red) is `gAudioThreadUpdateTimeTotalPerGfxTask`
second timer bar (yellow) is `gGfxTaskSentToNextReadyMinusAudioThreadUpdateTime`
(see `sSpeedMeterTimeEntryArray`)

you can see the red bar is not negligible (well this is emulator but I expect hardware to be similar at least in ratios)

Note: more or less, `gGraphUpdatePeriod = gGfxTaskSentToNextReadyMinusAudioThreadUpdateTime + gAudioThreadUpdateTimeTotalPerGfxTask + (waiting for task done and framebuffer swap)`

Reply:
I am not a fan of the super verbose names like this, but I do not have any suggestions, so I think we just go with this and improve it at some point if we can

Reply:
Yeah I guess we can do that tho it's a bit unfortunate